Automattic/wp-calypso
# P2 signup: add 'post-email confirmation' screen

#### Changes proposed in this Pull Request

* Part of the revised P2 signup flow project (p9lV3a-2MW-p2). 
* Signup flow: display success screen if user has verified their email during signup.
* See p9lV3a-2Lp-p2#comment-3599 for details about this step.


#### Testing instructions

* `ENABLE_FEATURES=no-force-sympathy yarn start`


1. Check that verified users are skipped over:
    a. While logged in as a verified user, visit `calypso.localhost:3000/start/p2-new`. Since your user account is already verified, you should go straight to the `p2-site` step, skipping the `p2-confirm-email` step.
2. Check that unverified users are made to go through the email verification step:
    a. In incognito mode, visit `calypso.localhost:3000/start/p2-new` and sign up for a new acccount, using an email address that you have access to.
    b. After signing up, you should get the "Check your email" step. Do not close this window.
    c. Copy the verification link and paste it in a browser tab, within the same incognito session. This should mark your test user account as verified, but will redirect to Calypso home. The redirect will be fixed in a separate PR.
    d. Return to the window in 2b and reload the page. You should get the "Email confirmed" notice. Click the "Continue" button and you should get the next step.



#### Screens
<img width="800" alt="Screen Shot 2021-12-28 at 4 10 27 PM" src="https://user-images.githubusercontent.com/730823/147544044-139ba95e-fcd7-4ca0-8edc-87ff2eaf947f.png">
